libraries/AP_HAL_Linux/Util_RPI.cpp

```diff
@@ -38,8 +38,16 @@ int UtilRPI::_check_rpi_version()
 
     FILE *f = fopen("/sys/firmware/devicetree/base/model", "r");
     if (f != nullptr && fgets(buffer, MAX_SIZE_LINE, f) != nullptr) {
-        int ret = sscanf(buffer + 12, "%d", &_rpi_version);
         fclose(f);
+        int ret = sscanf(buffer + 28, "%d", &_rpi_version);
```

Comment:
it would be very unfortunate if in this position it matched a number from a different string. Maybe just compare the string with strcmp()?

Reply:
the function's original logic assumes undetected RPI as RPI/2, also UtilRPI::get_rpi_version() callers assume it always return a valid value.

the newly added board uses strncmp() as suggested.